services/sharepoint-connector/deploy/Dockerfile

```diff
@@ -0,0 +1,67 @@
+# syntax=docker/dockerfile:1.7
+
+FROM node:22-bookworm AS base
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+
+FROM base AS pruner
+WORKDIR /app
+ARG APP_NAME
+RUN --mount=type=bind,source=.,target=/src,ro \
+    pnpm dlx turbo prune @unique-ag/${APP_NAME} --docker --cwd /src --out-dir /app/out
+
+FROM base AS deps
+WORKDIR /app
+
+COPY --from=pruner /app/out/json/ .
+COPY ./pnpm-lock.yaml ./pnpm-lock.yaml
+COPY ./pnpm-workspace.yaml ./pnpm-workspace.yaml
+
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
+    pnpm install --frozen-lockfile
+
+FROM base AS builder
+ARG APP_NAME
+WORKDIR /app
+
+COPY --from=deps /app/ .
+COPY ./tsconfig.json ./tsconfig.json
+COPY ./turbo.json ./turbo.json
+COPY --from=pruner /app/out/full/ .
+
+RUN pnpm --filter=@unique-ag/${APP_NAME} codegen
```

Comment:
Just a remark 
Although this might be required when we add codegen, at the moment we don't really have a codegen at this step so this codegen cmd doesn't exist yet. 
Also the exissting dockerfile has this codegen step so we should be fine keeping it here.